src/IRI.php

```diff
@@ -539,19 +540,22 @@ protected function remove_iunreserved_percent_encoded($match)
         // at the first byte!).
         $string = '';
         $remaining = 0;
+        $start = 0;
+        $character = 0;
+        $length = 0;
```

Comment:
Instead I would add extra `$valid = true` here to satisfy PHPStan. And add a comment that these variables will be initialized in the loop but PHPStan is not able to detect it ([currently](https://github.com/phpstan/phpstan/issues/8192)).

Reply:
done.